### PR TITLE
Ports: Add doxygen version 1.9.7

### PR DESCRIPTION
**Pull request by tbhaxor**

### CODE REVIEW

File: Ports/doxygen/package.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+)
+makeopts+=(
+    -Cbuild
+)
+installopts+=(
+    -Cbuild
+)
+depends=(
+    libiconv
+)
```
Comment (timschumi) [inline Ports/doxygen/package.sh:24]: Please adhere to the quoting as shown in the Ports README.

File: Ports/doxygen/patches/0001-Detect-serenity-os.patch [hunk 2]
```diff
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Gurkirat Singh <tbhaxor@gmail.com>
+Date: Mon, 21 Aug 2023 05:47:52 +0530
+Subject: [PATCH] Detect serenity os
```
Comment (timschumi) [inline Ports/doxygen/patches/0001-Detect-serenity-os.patch:4]:
```suggestion
Subject: [PATCH] Detect SerenityOS
```

Comment (timschumi) [reply Ports/doxygen/patches/0001-Detect-serenity-os.patch:4]: Also, looks like this is more of a "Pretend that SerenityOS is Linux-like"?

File: Ports/doxygen/package.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=doxygen
+version='1.9.7'
+files=(
+    "https://github.com/doxygen/doxygen/archive/refs/tags/Release_${version//./_}.tar.gz 691777992a7240ed1f822a5c2ff2c4273b57c1cf9fc143553d87f91a0c5970ee"
+)
+workdir="${port}-Release_${version//./_}"
+useconfigure='true'
+configopts=(
```
Comment (timschumi) [inline Ports/doxygen/package.sh:9]: This should presumably include `-DCMAKE_TOOLCHAIN_FILE=...`, which might also remove the need for some of the manual settings here.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/doxygen/patches/0001-Detect-serenity-os.patch (hunk 2 above)
Comment (tbhaxor) [reply Ports/doxygen/patches/0001-Detect-serenity-os.patch:4]: Yes, I have fixed this.

**Comment by tbhaxor** [comment]
Please confirm the updates, I will then squash them to one.

File: Ports/doxygen/package.sh (hunk 1 above)
Comment (tbhaxor) [reply Ports/doxygen/package.sh]: Why do we need to have quotes around depends, makeopts, port name there is no space between them?

Comment (timschumi) [reply Ports/doxygen/package.sh]: We just like to make it obvious to the reader that those are strings, and nothing else.

**Comment by timschumi** [comment]
Looks good, please squash the commits and we can get this merged.

**Comment by tbhaxor** [comment]
> Looks good, please squash the commits and we can get this merged.

Done.

File: Ports/doxygen/package.sh (hunk 1 above)
Comment (tbhaxor) [reply Ports/doxygen/package.sh]: Makes sense. I am working on shellcheck tested refactor of ports to include single or double quotes based on variable interpolation.

**Review by timschumi — CHANGES_REQUESTED** [review]
This does not install to the correct location:

```
[doxygen/install] -- Installing: /home/tim/dev/serenity/Build/x86_64/Root/home/tim/dev/serenity/Build/x86_64/Root/usr/local/bin/doxygen
```

You presumably need to override the `install` function to omit the `DESTDIR` setting.

**Comment by tbhaxor** [comment]
> This does not install to the correct location:
> 
> ```
> [doxygen/install] -- Installing: /home/tim/dev/serenity/Build/x86_64/Root/home/tim/dev/serenity/Build/x86_64/Root/usr/local/bin/doxygen
> ```
> 
> You presumably need to override the `install` function to omit the `DESTDIR` setting.

Where it should install? 

image

image

**Comment by timschumi** [comment]
> Where it should install?

`/usr/local/bin/doxygen` would be correct, but it doesn't seem to end up there in a clean build. Instead, a host path ends up inside of the SerenityOS disk image.

**Comment by tbhaxor** [comment]
It should be fixed now. Also I am now using `ninja` instead of `make`.

**Review by timschumi — APPROVED** [review]

**Pull request merged**